core/src/main/java/org/elasticsearch/common/network/NetworkModule.java

```diff
@@ -181,4 +175,39 @@ private void registerBuiltinAllocationCommands() {
     public boolean canRegisterHttpExtensions() {
         return transportClient == false;
     }
+
+    /**
+     * Registers a new {@link org.elasticsearch.transport.TransportService.TransportInterceptor}
+     */
+    public void addTransportInterceptor(TransportService.TransportInterceptor interceptor) {
```

Comment:
Is this just temporary, and can/should be replaced with a pull based interface in a followup?